xbmc/addons/kodi-addon-dev-kit/include/kodi/gui/controls/Edit.h

```diff
@@ -0,0 +1,278 @@
+  /// "edit control".
+  ///
+  /// @note The  call of the  control is only possible  from  the  corresponding
+  /// window as its class and identification number is required.
+  ///
+
+  //============================================================================
+  // see gui/definition.h for use of group "cpp_kodi_gui_controls_CEdit_Defs"
+  ///
+  /// \defgroup cpp_kodi_gui_controls_CEdit_Defs Definitions, structures and enumerators
+  /// \ingroup cpp_kodi_gui_controls_CEdit
+  /// @brief **Library definition values**
+  ///
+
+} /* namespace controls */
+} /* namespace gui */
+} /* namespace kodi */
+
+//============================================================================
+///
+/// \ingroup cpp_kodi_gui_controls_CEdit_Defs
+/// @{
+/// @anchor AddonGUIInputType
+/// @brief Text input types used on kodi::gui::controls::CEdit
+enum AddonGUIInputType
+{
+  /// Text inside edit control only readable
+  ADDON_INPUT_TYPE_READONLY = -1,
+  /// Normal text entries
+  ADDON_INPUT_TYPE_TEXT = 0,
+  /// To use on edit control only numeric numbers
+  ADDON_INPUT_TYPE_NUMBER,
+  /// To insert seconds
+  ADDON_INPUT_TYPE_SECONDS,
+  /// To insert time
+  ADDON_INPUT_TYPE_TIME,
+  /// To insert a date
+  ADDON_INPUT_TYPE_DATE,
+  /// Used for write in IP addresses
+  ADDON_INPUT_TYPE_IPADDRESS,
+  /// Text field used as password entry field with not visible text
+  ADDON_INPUT_TYPE_PASSWORD,
+  /// Text field used as password entry field with not visible text but
+  /// returned as MD5 value
+  ADDON_INPUT_TYPE_PASSWORD_MD5,
+  /// Use text field for search purpose
+  ADDON_INPUT_TYPE_SEARCH,
+  /// Text field as filter
+  ADDON_INPUT_TYPE_FILTER,
+  ///
+  ADDON_INPUT_TYPE_PASSWORD_NUMBER_VERIFY_NEW
+};
+/// @}
+//----------------------------------------------------------------------------
+
+namespace kodi
+{
+namespace gui
+{
+namespace controls
+{
+
+  class CEdit : public CAddonGUIControlBase
+  {
+  public:
+    //==========================================================================
+    ///
+    /// \ingroup cpp_kodi_gui_controls_CEdit
+    /// @brief Construct a new control
+    ///
+    /// @param[in] window               related window control class
+    /// @param[in] controlId            Used skin xml control id
+    ///
+    CEdit(CWindow* window, int controlId)
+      : CAddonGUIControlBase(window)
+    {
+      m_controlHandle = m_interface->kodi_gui->window->get_control_edit(m_interface->kodiBase, m_Window->GetControlHandle(), controlId);
+      if (!m_controlHandle)
+        kodi::Log(ADDON_LOG_FATAL, "kodi::gui::control::CEdit can't create control class from Kodi !!!");
+    }
+    //--------------------------------------------------------------------------
+
+    //==========================================================================
+    ///
+    /// \ingroup cpp_kodi_gui_controls_CEdit
+    /// @brief Destructor
+    ///
+    virtual ~CEdit()
+    {
+    }
```

Comment:
One of my favorites (as you know): "virtual ~CEdit() = default;" ? Here and everywhere in this PR. ;-)